### PR TITLE
Drop ForeignKeys from TagKey, TagValue and GroupTagKey.

### PR DESCRIPTION
**Pull request by bretthoerner**
Like https://github.com/getsentry/sentry/commit/27f7e2d085368605738c955b3a1b31635174b27a this has an empty/fake migration to appease South.

This doesn't handle any use of joins (I want to do that in another PR -- if there even are any) so this only allows us to drop the constraints and keep it on the same database.

Note that we'll presumably want to drop FK constraints on production after this is deployed (?).

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR includes migrations</td>
    </tr>
  </tbody>
</table>

## Migration Checklist

- [ ] new columns need to be nullable (unless table is new)
- [ ] migration with any new index needs to be done concurrently
- [ ] data migrations should not be done inside a transaction
- [ ] before merging, check to make sure there aren't conflicting migration ids


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/models/grouptagkey.py [hunk 1]
```diff
@@ -23,8 +23,8 @@ class GroupTagKey(Model):
     """
     __core__ = False
 
-    project = FlexibleForeignKey('sentry.Project', null=True)
-    group = FlexibleForeignKey('sentry.Group')
+    project_id = BoundedPositiveIntegerField(null=True)
```
Comment (mattrobenolt) [inline src/sentry/models/grouptagkey.py]: These now need normal `db_index=True` on them. Since we don't want to lose an index.

File: src/sentry/models/tagkey.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ class TagKey(Model):
     """
     __core__ = False
 
-    project = FlexibleForeignKey('sentry.Project')
+    project_id = BoundedPositiveIntegerField()
```
Comment (mattrobenolt) [inline src/sentry/models/tagkey.py]: `db_index=True` here too.

File: src/sentry/models/tagvalue.py [hunk 3]
```diff
@@ -24,7 +24,7 @@ class TagValue(Model):
     """
     __core__ = False
 
-    project = FlexibleForeignKey('sentry.Project', null=True)
+    project_id = BoundedPositiveIntegerField(null=True)
```
Comment (mattrobenolt) [inline src/sentry/models/tagvalue.py]: same

File: src/sentry/receivers/releases.py [hunk 4]
```diff
@@ -16,23 +16,25 @@ def ensure_release_exists(instance, created, **kwargs):
     if instance.data and instance.data.get('release_id'):
         return
 
+    project = Project.objects.get(pk=instance.project_id)
```
Comment (mattrobenolt) [inline src/sentry/receivers/releases.py]: What's the point in doing this? Why was wrong with just `instance.project` ? These changes seem entirely unrelated.

**Review by mattrobenolt — CHANGES_REQUESTED** [review]

File: src/sentry/models/grouptagkey.py (hunk 1 above)
Comment (bretthoerner) [reply src/sentry/models/grouptagkey.py]: Good catch, these aren't on `GroupTagValue` either though, should I add them?

Comment (mattrobenolt) [reply src/sentry/models/grouptagkey.py]: Yeah, that's my bad then.

It technically shouldn't change anything since an earlier migration of adding the table would have created it correctly. Adding it here is effectively just bookkeeping.

File: src/sentry/receivers/releases.py (hunk 4 above)
Comment (bretthoerner) [reply src/sentry/receivers/releases.py]: `instance` is a `TagValue`,

```
project_id = BoundedPositiveIntegerField(db_index=True, null=True)
```

Maybe I'm missing something?

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR includes migrations</td>
    </tr>
  </tbody>
</table>

## Migration Checklist

- [ ] new columns need to be nullable (unless table is new)
- [ ] migration with any new index needs to be done concurrently
- [ ] data migrations should not be done inside a transaction
- [ ] before merging, check to make sure there aren't conflicting migration ids


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/receivers/releases.py (hunk 4 above)
Comment (mattrobenolt) [reply src/sentry/receivers/releases.py]: Ohh, you're right. I read this a bit too quickly. Thought it was a `Release`. Love types.

Comment (bretthoerner) [reply src/sentry/receivers/releases.py]: I actually renamed all the `instance` params because it was annoying me, and then I realized (via tests) that we/Django calls it with kwargs (`instance=foo`). Love Python.

**Review by mattrobenolt — APPROVED** [review]
wcgw 🦄
